### ISSUE TITLE
Getting the error "expression did not evaluate to a constant" in VS 2015.

### ISSUE THREAD

**Issue by a52siddi** [davisking/dlib#268]
Hello.
I am following the compilation procedure that has been posted on the dlib website.

I created an empty project called "bayes". I right-clicked on "Source Files", went to "Existing Item...", then added "source.cpp" to the "Source Files". I saw another question on the forum which stated the same error:
https://github.com/davisking/dlib/issues/194

I went to github and copied and pasted the code for source.cpp and matrix_utilities.h (The error is in this file) from github in my project. Despite that, I have been getting the same error when I run my code.

I right-clicked on the name of the project (bayes), clicked on "Properties", then went to "VC++ Directories" and then in "Include Directories" I added the path of the dlib-19.1 folder.
I even disabled the "Precompiled Headers" option under the "C/C++" heading. I don't understand why I am not able to run my code.

I compiled all the examples using CMake. I am able to open the examples.sln file and run most of the project files present there.

Why can't I run the code when I create an empty project, include all directories and copy and paste the same code in that project?

Waiting eagerly for your response.

**Comment by davisking** [comment]
Do not spam questions to every possible dlib forum.  It does not increase the probability of getting a useful response.  

For your case, you are asking about how to setup a basic C++ project in visual studio, not about dlib.  Since this question has nothing to do with dlib you should ask it elsewhere.  I would suggest finding some forum about visual studio.

**Comment by a52siddi** [comment]
Hello Mr.King.

I will be mindful of that in the future. Excuse me for any inconvenience this may have caused you.

My question is very similar to one of the questions in your forum. I had posted the link for that question in my query. I followed your answer to that question but my error wasn't resolved.

The question I am referring to is in the following link:
https://github.com/davisking/dlib/issues/194

I'll greatly appreciate it if you reconsider my question and help me resolve my problem.

Regards,
Abdullah.

Sent from Yahoo Mail on Android
